### DISCUSSION TITLE
Edit actions-runner-controller default settings

### DISCUSSION THREAD

**Discussion by klepiz** [actions/actions-runner-controller#1994] [Ideas]
Is it possible to change amount of replica sets for actions-runner-controller for k8s? By default it will create only 1 replica set, but I would like to have more than that. 
![Screen Shot 2022-11-07 at 11 06 50](https://user-images.githubusercontent.com/56735755/200371301-41379874-ef0b-4236-8201-9eaddafb18ba.png)

I currently have 2 k8s nodes, but the actions-runner-controller its only on one of it.

**Comment by mumoshu** [comment]
Can you just set the `replicas` field to 2 or more? 🤔 
(Well anyway, let me convert this into a Discussion as it doesn't look like an actionable feature request atm

**Reply by klepiz** [reply]
yes, that works but still, all the pods have been created on a single node

**Reply by toast-gear** [reply]
use `topologySpreadConstraints:` to control the distribution

**Reply by amalberty** [reply]
Hello @toast-gear,
Can you please provide some guidance on how to use topologySpreadConstraints?
Thank you.

**Reply by dongho-jung** [reply]
@amalberty refer to [here](https://github.com/actions-runner-controller/actions-runner-controller/tree/master/charts/actions-runner-controller)

you can control the distribution with topologySpreadConstraints as follows:

values.yaml
```yaml
...
topologySpreadConstraints: ...
...
```

You can get info about topologySpreadConstraints from [here](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/)

**Reply by amalberty** [reply]
Thank you

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
